cajita/src/Cajita_ParticleGridDistributor.hpp

```diff
@@ -0,0 +1,390 @@
+/****************************************************************************
+ * Copyright (c) 2018-2021 by the Cabana authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Cabana library. Cabana is distributed under a   *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef CABANA_PARTICLEGRIDDISTRIBUTOR_HPP
+#define CABANA_PARTICLEGRIDDISTRIBUTOR_HPP
+
+#include <Cabana_DeepCopy.hpp>
+#include <Cabana_Distributor.hpp>
+
+#include <Cajita_GlobalGrid.hpp>
+#include <Cajita_GlobalMesh.hpp>
+#include <Cajita_LocalGrid.hpp>
+#include <Cajita_LocalMesh.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <vector>
+
+namespace Cajita
+{
+
+//---------------------------------------------------------------------------//
+// Particle Grid Distributor
+//---------------------------------------------------------------------------//
+/*!
+  \brief Determine which particles need to be migrated using the grid
+
+  \tparam LocalGridType Cajita LocalGrid type.
+*/
+template <class LocalGridType>
+class ParticleGridDistributor
```

Comment:
Does this even need to be a class? Based on your implementations below the content of these could just be in those free functions.

Reply:
I converted this to a class to make the variable dimension easier. Is there a good way to do that otherwise? I'm okay with moving this to `Impl` if we keep it

Reply:
I fixed this as you suggested